File: Heat/Repositories/HeatDBContext.vb
Imports System.Data.Entity
Imports Heat.Models
Imports Microsoft.AspNet.Identity
Imports Microsoft.AspNet.Identity.EntityFramework
Imports log4net


Namespace Repositories

    Public Class HeatDBContext
        Inherits IdentityDbContext(Of HeatUser)
        Implements IHeatDBContext

        Private _logger As ILog

        Sub New()
            MyBase.New("DefaultConnection")
            'Entity.Database.SetInitializer(Of HeatDBContext)(New DropCreateDatabaseIfModelChanges(Of HeatDBContext))
            'Entity.Database.SetInitializer(Of HeatDBContext)(Nothing)
            'Entity.Database.SetInitializer(Of HeatDBContext)(New HeatDBInitializer)
            'Me.Configuration.LazyLoadingEnabled = True

            '_logger = LogManager.GetLogger(GetType(HeatDBContext))
            '_logger.Debug("HeatDbContext created")

        End Sub


        Public Shared Function Create() As HeatDBContext
            Return New HeatDBContext()
        End Function

        Public Sub SetModified(entity As Object) Implements IHeatDBContext.SetModified
            Entry(entity).State = EntityState.Modified
        End Sub

        Property Customers As DbSet(Of Customer) Implements IHeatDBContext.Customers
        Property Address As DbSet(Of Address) Implements IHeatDBContext.Addresses
        Property AddressTypes As DbSet(Of AddressType) Implements IHeatDBContext.AddressTypes
        Property Contacts As DbSet(Of Contact) Implements IHeatDBContext.Contacts
        Property WarehouseMovement As DbSet(Of WarehouseMovement) Implements IHeatDBContext.WarehouseMovements
        Property Plants As DbSet(Of Plant) Implements IHeatDBContext.Plants
        Property PlantTypes As DbSet(Of PlantType) Implements IHeatDBContext.PlantTypes
        Property PlantClasses As DbSet(Of PlantClass) Implements IHeatDBContext.PlantClasses
        Property Invoices As DbSet(Of Invoice) Implements IHeatDBContext.Invoices
        Property InvoiceRows As DbSet(Of InvoiceRow) Implements IHeatDBContext.InvoiceRows
        Property Payments As DbSet(Of Payment) Implements IHeatDBContext.Payments
        Property Fuels As DbSet(Of Fuel) Implements IHeatDBContext.Fuels
        Property CausalDocuments As DbSet(Of CausalDocument)
        Property Numberings As DbSet(Of Numbering) Implements IHeatDBContext.Numberings
        Property DocumentTypes As DbSet(Of Models.DocumentType) Implements IHeatDBContext.DocumentTypes
        Property SerialSchemes As DbSet(Of Models.SerialScheme) Implements IHeatDBContext.SerialSchemes
        Property Warehouses As DbSet(Of Warehouse) Implements IHeatDBContext.Warehouses
        Property CausalWarehouseGroups As DbSet(Of CausalWarehouseGroup) Implements IHeatDBContext.CausalWarehouseGroups
        Property CausalWarehouses As DbSet(Of CausalWarehouse) Implements IHeatDBContext.CausalWarehouses
        Property Seller As DbSet(Of Seller) Implements IHeatDBContext.Sellers
        Property Products As DbSet(Of Product) Implements IHeatDBContext.Products
        Property DescriptiveInvoiceRows As DbSet(Of DescriptiveInvoiceRow) Implements IHeatDBContext.DescriptiveInvoiceRows
        Property ProductInvoiceRows As DbSet(Of ProductInvoiceRow) Implements IHeatDBContext.ProductInvoiceRows
        Property ActionTypes As DbSet(Of Models.ActionType) Implements IHeatDBContext.ActionTypes
        Property WorkOperators As DbSet(Of Models.WorkOperator) Implements IHeatDBContext.WorkOperators
        Property WorkActions As DbSet(Of WorkAction) Implements IHeatDBContext.WorkActions
        Property Operations As DbSet(Of Operation) Implements IHeatDBContext.Operations
        Property Manifacturers As DbSet(Of Manifacturer) Implements IHeatDBContext.Manifacturers
        Property ManifacturerModels As DbSet(Of ManifacturerModel) Implements IHeatDBContext.ManifacturerModels
        Property HeatTransferFluids As DbSet(Of HeatTransferFluid) Implements IHeatDBContext.HeatTransferFluids
        Property ThermalUnits As DbSet(Of Models.ThermalUnit) Implements IHeatDBContext.ThermalUnits
        Property ThermalUnitKinds As DbSet(Of ThermalUnitKind) Implements IHeatDBContext.ThermalUnitKinds
        Property Heaters As DbSet(Of Heater) Implements IHeatDBContext.Heaters
        Property PlantServices As DbSet(Of Models.PlantService) Implements IHeatDBContext.PlantServices

        Property Media As DbSet(Of Medium) Implements IHeatDBContext.Media

        Public Overrides Function SaveChanges() As Integer Implements IHeatDBContext.SaveChanges
            Return MyBase.SaveChanges
        End Function

    End Class

End Namespace

File: Heat.ConvertedToC#/Repositories/HeatDBContext.cs
using System.Data.Entity;
using Heat.Models;
using Microsoft.AspNet.Identity.EntityFramework;
using log4net;
using System.Diagnostics;

namespace Heat.Repositories
{

    public class HeatDBContext : IdentityDbContext<HeatUser>, IHeatDBContext
	{


		private ILog _logger;
		public HeatDBContext() : base("DefaultConnection")
		{
            //Entity.Database.SetInitializer(Of HeatDBContext)(New DropCreateDatabaseIfModelChanges(Of HeatDBContext))
            //Entity.Database.SetInitializer(Of HeatDBContext)(Nothing)
            //Entity.Database.SetInitializer(Of HeatDBContext)(New HeatDBInitializer)
            //Me.Configuration.LazyLoadingEnabled = True

            //_logger = LogManager.GetLogger(GetType(HeatDBContext))
            //_logger.Debug("HeatDbContext created")
            Database.Log = s => Debug.WriteLine(s);

		}


		public static HeatDBContext Create()
		{
			return new HeatDBContext();
		}

		public void SetModified(object entity)
		{
			Entry(entity).State = EntityState.Modified;
		}

		public DbSet<Customer> Customers { get; set; }
		public IDbSet<Address> Addresses { get; set; }
			
		public IDbSet<AddressType> AddressTypes { get; set; }
		public DbSet<Contact> Contacts { get; set; }

		public DbSet<Plant> Plants { get; set; }
		public DbSet<PlantType> PlantTypes { get; set; }
		public DbSet<PlantClass> PlantClasses { get; set; }
		public IDbSet<Invoice> Invoices { get; set; }
		public IDbSet<InvoiceRow> InvoiceRows { get; set; }
		public IDbSet<Payment> Payments { get; set; }
		public DbSet<Fuel> Fuels { get; set; }
		public IDbSet<CausalDocument> CausalDocuments { get; set; }
		public DbSet<Numbering> Numberings { get; set; }
		public IDbSet<Models.DocumentType> DocumentTypes { get; set; }
		public IDbSet<Models.SerialScheme> SerialSchemes { get; set; }
		public IDbSet<Warehouse> Warehouses { get; set; }
		public IDbSet<CausalWarehouseGroup> CausalWarehouseGroups { get; set; }
		public IDbSet<CausalWarehouse> CausalWarehouses { get; set; }
		

		public IDbSet<Product> Products { get; set; }
		public IDbSet<DescriptiveInvoiceRow> DescriptiveInvoiceRows { get; set; }
		public IDbSet<ProductInvoiceRow> ProductInvoiceRows { get; set; }
		public IDbSet<Models.ActionType> ActionTypes { get; set; }
		public IDbSet<Models.WorkOperator> WorkOperators { get; set; }
		public IDbSet<WorkAction> WorkActions { get; set; }
		public IDbSet<Operation> Operations { get; set; }
		public DbSet<Manifacturer> Manifacturers { get; set; }
		public DbSet<ManifacturerModel> ManifacturerModels { get; set; }
		public IDbSet<HeatTransferFluid> HeatTransferFluids { get; set; }
		public DbSet<Models.ThermalUnit> ThermalUnits { get; set; }
		public IDbSet<ThermalUnitKind> ThermalUnitKinds { get; set; }
		public IDbSet<Heater> Heaters { get; set; }
		public IDbSet<Models.PlantService> PlantServices { get; set; }

		public IDbSet<Medium> Media { get; set; }
		public IDbSet<OutboundCall> OutboundCalls { get; set; }
		public  DbSet<ProposedOutBoundCall> ProposedOutboundCalls { get; set; }
        public  DbSet<AssignedOutboundCall> AssignedOutboundCalls { get; set; }
        public  DbSet<ProposedCallsGeneration> ProposedCallsGenerations {get; set;}

        public IDbSet<WarehouseMovement> WarehouseMovements { get; set; }
        
        public DbSet<PlantRole> PlantRoles { get; set; }

        public IDbSet<Seller> Sellers { get; set; }

        public override int SaveChanges()
		{
			return base.SaveChanges();
		}

        
    }

}


